src/formats/onenote.ts

```diff
@@ -425,39 +464,54 @@ export class OneNoteImporter extends FormatImporter {
 
 	// Convert OneNote styled elements to valid HTML for proper htmlToMarkdown conversion
 	styledElementToHTML(pageElement: HTMLElement): HTMLElement {
-		const styledElements = pageElement.querySelectorAll('[style]');
-
-		// For some reason cites/quotes are not converted into Markdown (possible htmlToMarkdown bug), so we do it ourselves temporarily
-		const cites = pageElement.findAll('cite');
-		cites.forEach((cite) => cite.innerHTML = '> ' + cite.innerHTML + '<br>');
-
 		// Map styles to their elements
 		const styleMap: { [key: string]: string } = {
 			'font-weight:bold': 'b',
 			'font-style:italic': 'i',
 			'text-decoration:underline': 'u',
 			'text-decoration:line-through': 's',
 			'background-color': 'mark',
-			'font-family:Consolas': 'pre',
 		};
+		// Cites/quotes are not converted into Markdown (possible htmlToMarkdown bug?), so we do it ourselves temporarily
+		const cites = pageElement.findAll('cite');
+		cites.forEach((cite) => cite.innerHTML = '> ' + cite.innerHTML + '<br>');
+		
+		// Convert preformatted text into code blocks
+		let inCodeBlock: boolean = false;
+		let codeElement: HTMLElement = document.createElement('pre');
 
-		styledElements.forEach(element => {
+		const elements = pageElement.querySelectorAll('*');
+		elements.forEach(element => {
 			const style = element.getAttribute('style') || '';
 			const matchingStyle = Object.keys(styleMap).find(key => style.includes(key));
 
-			if (matchingStyle) {
-				const newElementTag = styleMap[matchingStyle];
-				const newElement = document.createElement(newElementTag);
+			if (style?.contains('font-family:Consolas')) {
+				if (!inCodeBlock) {
+					inCodeBlock = true;
+					element.replaceWith(codeElement);
+					codeElement.innerHTML = '```\n' + element.innerHTML + '\n```';
+				}
+				else {
+					element.remove();
+					// Append the content and add fences in case there's no next element
+					codeElement.innerHTML = codeElement.innerHTML.slice(0, -3) + element.innerHTML + '\n```';
+				}
+			}
+			else if (element.nodeName === 'BR' && inCodeBlock) {
+				codeElement.innerHTML = codeElement.innerHTML.slice(0, -3) + '\n```';
+				element.remove();
+			}
+			else {
+				if (inCodeBlock) inCodeBlock = false; codeElement = document.createElement('pre');
```

Comment:
Please reformat this into a block with curly braces.

Reply:
fixed!